### DISCUSSION TITLE
Increase minimum JDK version to support

### DISCUSSION THREAD

**Discussion by oliemansm** [graphql-java-kickstart/graphql-java-servlet#487] [Polls]
Java 17 has been released quite some time ago and is the latest stable release replacing Java 8 as LTS. Although Java 8 in practice is still heavily used, it's unclear what the usage is of applications relying on this library. The latest version of Spring Boot v3.0.0 has changed it's support and now requires at least Java 17.

Supporting older versions of Java becomes more and more challenging and ideally we can increase the minimum JDK version to support, ideally even to v17.

**Comment by federicorispo** [comment]
@oliemansm I am late for the voting :cry: but I have an idea.

First some facts:
 - Spring will continue to maintain the 5.3.* release [until the end of 2025](https://spring.io/projects/spring-framework#support). This release is the last one supporting versions of JDK < 17 and using the javax dependency
 - RESTEasy (I choose it because I use it for some projects) jumped on jakarta a year ago with the 4.* version but they decided to support jdk versions >= 11 even for their last version (6.*)
 - graphql-java team supports jdk versions >= 11 ([reference](https://www.graphql-java.com/blog/java-11-required))
 
  
I am working on some (non spring) projects where I am forced to use the openjdk 11/13 and with the last release of graphql-java-servlet I am (and I will be) stucked forever with the previous version (14.0.0) that will not be updated with the newer versions of graphql-java and I cannot migrate from javax to jakarta.
I think there are a lot of legacy projects out there having the same issue, so dropping the support of jdk 11 could cause some headache for the maintainers of these projects.

So I have a suggestion (like graphql-java does):
 - create a branch `14.x.x-jakarta` where the graphql-java-servlet uses jakarta dependencies but it support jdk 11: this branch should cover the case on which the developer is stuck with the older version of the jdk but he/she did not use Spring so he/she could start using jakarta (instead of javax). This branch will not contain all the tests related to spring for obvious reasons.
 - create a branch `14.x.x-javax` where the graphql-java-servlet continue to use javax and continue to support jdk 11: this branch should cover the case on which the developer is using spring 5.3.* but want to use the latest version of graphql-java.

These branches could be aligned with the main branch every time a new release is made so the effort of cherry-picking the new commits is low.

I am happy to help maintaining these branches if you agree to try this solution. (I already have a commit on my fork for the second branch and I will create another one for the first one)

**Reply by oliemansm** [reply]
If you're willing to help maintain than I'm all for it.

**Reply by federicorispo** [reply]
@oliemansm Great news! Can you please create the two branches so I can open PRs?

**Comment by federicorispo** [comment]
@oliemansm Sorry to bother you again, did you have time to think how to approach the proposed solution and if the flow using the PR to the two branches is feasable to maintain the two different versions of graphql-java-servlet?

**Reply by oliemansm** [reply]
@federicorispo Apologies, pretty busy period. Could you join our slack so we can communicate in a more direct way? Also seeing you're going to be more involved maintaining those branches. https://join.slack.com/t/graphqljavakickstart/shared_invite/zt-1r0rm7kxv-TneVp824seSS1Ul3YAB~Mg

**Comment by AdamKossemtini** [comment]
@oliemansm I have a spring boot 2.7 application using   com.graphql-java-kickstart:graphql-spring-boot-starter V11.1.0,  I want to migrate it to spring boot 3.0.7 using com.graphql-java-kickstart:graphql-spring-boot-starter V15.0.0, what are the steps that I need to follow to get the compatibility, (I tried to upgrade the spring boot and GraphQL to 15.0.0 but I got null data response)
